modules/mod-deep-learning/DeepLearningEffectBase.cpp

```diff
@@ -0,0 +1,363 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   DeepLearningEffectBase.cpp
+   Hugo Flores Garcia
+
+******************************************************************/
+
+#include "DeepLearningEffectBase.h"
+
+#include "DeepModelManager.h"
+#include "ModelManagerPanel.h"
+
+#include "FileNames.h"
+#include "Shuttle.h"
+#include "ShuttleGui.h"
+#include <wx/range.h>
+
+#include <torch/script.h>
+
+#include <wx/log.h>
+#include <wx/stattext.h>
+
+#include <WaveClip.h>
+
+// ModelCardPanel
+
+DeepLearningEffectBase::DeepLearningEffectBase()
+{
+   EnablePreview(false);
+}
+
+bool DeepLearningEffectBase::Init()
+{
+   // Catch any errors while setting up the DeepModelManager
+   try 
+   {
+      DeepModelManager &manager = DeepModelManager::Get();
+      
+      // try loading the model (if available)
+      mModel = std::make_shared<DeepModel>();
+      return true;
+   }
+   catch (const InvalidModelCardDocument &e)
+   {
+      Effect::MessageBox(XO("Error initalizing the Model Manager %s.").Format(e.what()),
+      wxICON_ERROR);
+      return false;
+   }
+}
+
+void DeepLearningEffectBase::End()
+{
```

Comment:
This function executes inside a destructor (a "finally" block in Effect.cpp) and should be guaranteed not to let exceptions escape.  Can I be sure of that, for the following steps?

You could wrap all in `GuardedCall<void>([&]{ ... });`

Which just eats the exceptions, not handling them, but if exceptions are of subclasses of `MessageBoxException`, then a delayed action is enqueued to the event loop, to show the user a message later.

In fact, maybe you can simplify away other catch blocks in this project, relying on `SimpleMessageBoxException` to notify the user.